pkg/skaffold/config/util.go

```diff
@@ -214,23 +210,46 @@ func GetDebugHelpersRegistry(configFile string) (string, error) {
 	return constants.DefaultDebugHelpersRegistry, nil
 }
 
-func isDefaultLocal(kubeContext string, detectMinikubeCluster bool) bool {
+// IsImageLoadingRequired checks if the cluster requires loading images into it
+func IsImageLoadingRequired(configFile string) (bool, error) {
+	cfg, err := GetConfigForCurrentKubectx(configFile)
+	if err != nil {
+		return false, err
+	}
+
+	kubeContext := cfg.Kubecontext
+	kindDisableLoad := cfg.KindDisableLoad != nil && *cfg.KindDisableLoad
+	k3dDisableLoad := cfg.K3dDisableLoad != nil && *cfg.K3dDisableLoad
+
+	return (IsKindCluster(kubeContext) && !kindDisableLoad) ||
+		(IsK3dCluster(kubeContext) && !k3dDisableLoad), nil
+}
+
+func isDefaultLocal(configFile string, detectMinikubeCluster bool) (bool, error) {
+	cfg, err := GetConfigForCurrentKubectx(configFile)
+	if err != nil {
+		return false, err
+	}
+
+	kubeContext := cfg.Kubecontext
+
+	imageLoadingRequired, err := IsImageLoadingRequired(configFile)
```

Comment:
`isDefaultLocal` seems to be trying to guess if it's a local cluster based on heuristics. I think we shouldn't conflate `imageLoadingRequired` into this. This eventually affects the value of `pushImages` in [types.go](https://github.com/GoogleContainerTools/skaffold/blob/2cb8d2e4663c1d42383ca8d56de6816fc8e4cbce/pkg/skaffold/build/local/types.go#L96). 
Perhaps we continue specifying:
```
build:
  local:
    push: true
```
when we want to push the image. 

This fix can be scoped to allow mechanism for disabling loading images https://github.com/GoogleContainerTools/skaffold/issues/3855#issuecomment-614166631

Reply:
I'm worried about developer experience here - they need to remember to modify skaffold.yaml when using *-disable-load.
I've reverted changes to isDefaultLocal and moved the call to [types.go](https://github.com/GoogleContainerTools/skaffold/pull/5012/files#diff-e66745249f227bd6e963daceb7e6e7ab71fd8b746eaead3e547d7da08a0a413fR102).